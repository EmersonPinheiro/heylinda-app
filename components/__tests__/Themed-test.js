import * as React from 'react'
import { render } from '@testing-library/react-native'
import { Text, View, useThemeColor } from '../Themed'
import Colors from '../../constants/Colors'

const mockTheme = 'dark'

jest.mock('../../hooks/useColorScheme', () => {
  return jest.fn(() => ({
    useColorScheme: () => mockTheme,
  }))
})

describe('Themed component tests', () => {
  it('renders themed text correctly', () => {
    const tree = render(<Text />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders themed View correctly', () => {
    const tree = render(<View />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('use correct dark theme color for text', () => {
  //   const colorName = 'text'
  //   const color = useThemeColor({}, colorName)
  //   expect(color).toBe(Colors.dark[colorName])
  // })
})
