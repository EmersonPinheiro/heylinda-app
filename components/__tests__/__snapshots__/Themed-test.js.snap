// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themed component tests renders themed View correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
      },
      undefined,
    ]
  }
/>
`;

exports[`Themed component tests renders themed text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      undefined,
    ]
  }
/>
`;
